compiler/damlc/daml-stdlib-src/DA/Internal/Prelude.daml

```diff
@@ -561,3 +561,6 @@ sum = foldl (+) aunit
 -- | Calculate the product over all elements
 product : Multiplicative a => [a] -> a
 product = foldl (*) munit
+
+undefined : a
+undefined = error "undefined"
```

Comment:
Let's not add this but rather replace the use sites by stuff like `magic @"exercise"` and `magic @"archive"`. If that breaks stuff because we don't erase all the `magic`s during conversion to DAML-LF, let's fix that.